src/java/org/apache/cassandra/cql3/functions/AbstractFunction.java

```diff
@@ -45,7 +45,7 @@ protected AbstractFunction(FunctionName name, List<AbstractType<?>> argTypes, Ab
     {
         this.name = name;
         this.argTypes = argTypes;
-        this.returnType = returnType;
+        this.returnType = returnType != null ? returnType.unwrap() : null;
```

Comment:
I think the return type should never be null, so we might as well throw an assertion, but no need to do it now.

Reply:
Yeah, I was thinking the same. Leaving it like this for now, also easier during rebase with Apache